scripts/subspace_mlp_demo.py

```diff
@@ -0,0 +1,191 @@
+# This demo replicates Figure 2 of the paper
+# "Measuring the Intrinsic Dimension of Objetive Landscape"
+# By Li et al. (https://arxiv.org/abs/1804.08838)
+# We consider a 2-layer MLP with ReLU activations
+# Code based on the following repos:
+# * https://github.com/ganguli-lab/degrees-of-freedom
+# * https://github.com/uber-research/intrinsic-dimension
+
+# Author: Gerardo Durán-Martín (@gerdm)
+
+import superimport
+
+import jax
+import optax
+import jax.numpy as jnp
+import flax.linen as nn
+import numpy as np
+import tensorflow_datasets as tfds
+import matplotlib.pyplot as plt
+import pyprobml_utils as pml
+from jax import random
+from time import time
+from functools import partial
+
+
+def get_datasets():
+    """
+    Load MNIST train and test datasets into memory
+    """
+    ds_builder = tfds.builder("mnist")
+    ds_builder.download_and_prepare()
+    train_ds = tfds.as_numpy(ds_builder.as_dataset(split="train", batch_size=-1))
+    test_ds = tfds.as_numpy(ds_builder.as_dataset(split="test", batch_size=-1))
+    train_ds["image"] = jnp.float32(train_ds["image"]) / 255.
```

Comment:
Can you name the fields X and y rather than image and label, to make it more generic?